PropTechAPI - コピー/src/main/java/com/PropTech/UserRegist/PropTechAppAPI/PropTechAPI/Logic/UserRegistrationLogic.java

```diff
@@ -0,0 +1,213 @@
+		responsedetails.setMessage(responsecommon.getResponseMessage());
+		return Response.status(Response.Status.CREATED.getStatusCode()).entity(responsedetails).build();
+	}
+	
+	private Response RequestBodyCheck(Request request, RequestBody body) {
+		ResponseDetails responsedetails = new ResponseDetails();
+		if(stringutil.isNullorEmpty(body.getsex()) ||
+		   stringutil.isNullorEmpty(body.getage()) ||
+		   stringutil.isNullorEmpty(body.getprofession()) ||
+		   stringutil.isNullorEmpty(body.getusername()) ||
+		   stringutil.isNullorEmpty(body.getpassword()) ||
+		   stringutil.isNullorEmpty(body.gettoken())) {
+		   return GenerateResponse(ResponseCommon.E400);
+		}else {
+			//System.out.println("Nullはありませんでした。");
+		}
+		return Response.status(Response.Status.CREATED.getStatusCode()).entity(responsedetails).build();
+	}
+	
+	private Response GenerateddbConnectionGetseqId(StringBuilder seqID) {
+		ResponseDetails responsedetails = new ResponseDetails();
+		String Format = new String();
+		
+		try(Connection connection = DriverManager.getConnection(URL, USER, PASSWORD);
+				PreparedStatement preparedStatement = connection.prepareStatement("SELECT NEXTVAL('apiseqno')");
+					ResultSet resultset = preparedStatement.executeQuery();){
+			boolean br = false;
+			br = resultset.next();
+			
+			Date date = new Date();
+			SimpleDateFormat dateformat = new SimpleDateFormat("yyyyMMdd");
+			String stringDate = dateformat.format(date);
+			Format = stringDate + String.format("%07d", resultset.getInt("nextval"));
+			seqID.append(Format);
+		}catch(SQLException exception) {
+			exception.printStackTrace();
+			//System.out.println(exception);
+			return GenerateResponse(ResponseCommon.E500);
+		}finally {
+			//System.out.println("完了");
+		}
+		return Response.status(Response.Status.CREATED.getStatusCode()).entity(responsedetails).build();
+	}
+	
+	private Response GeneratedHashPassword(RequestBody body, StringBuilder HashPassword) {
+		ResponseDetails responsedetails = new ResponseDetails();
+		String SHA256Password = PasswordHashUtil.getSaftyPassword("password", body.getpassword());
+		HashPassword.append(SHA256Password);
+		return Response.status(Response.Status.CREATED.getStatusCode()).entity(responsedetails).build();
+	}
+	
+	private Response GeneratedbConnectionSelect(StringBuilder HashPassword) {
+		ResponseDetails responsedetails = new ResponseDetails();
+		try(Connection connection = DriverManager.getConnection(URL, USER, PASSWORD);
+				PreparedStatement preparedStatement = connection.prepareStatement("SELECT * FROM userregistrationapi WHERE password = ?")){
+			preparedStatement.setString(1, HashPassword.toString());
+			
+			try (ResultSet resultset = preparedStatement.executeQuery()){
+				while(resultset.next()) {
+					System.out.println(resultset.getString("password"));
+					if(resultset.getString("password").equals(null)) {
+						//Not Execute
+					}else {
+						return GenerateResponse(ResponseCommon.E500);

```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the resource returned by the following method call: *GenerateResponse*. The resource is returned at line 154. There are other execution paths that don&#39;t close the resource or return it. Ensure the object returned by *GenerateResponse()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).